shared/test_run_filter.go

```diff
@@ -15,10 +15,27 @@ import (
 	mapset "github.com/deckarep/golang-set"
 )
 
+// SHAs is a helper type for a slice of commit/revision SHAs.
+type SHAs []string
+
+// EmptyOrLatest returns whether the shas slice is empty, or only contains
+// one item, which is the latest keyword.
+func (s SHAs) EmptyOrLatest() bool {
+	return len(s) < 1 || len(s) == 1 && IsLatest(s[0])
+}
+
+// FirstOrLatest returns the first sha in the slice, or the latest keyword.
+func (s SHAs) FirstOrLatest() string {
```

Comment:
FWIW, at the call sites I first found this name odd, like "first or last", but I don't know how to improve it without changing what the keyword is.